### PR TITLE
Support displaying options on focus

### PR DESCRIPTION
**Pull request by hcliff**
Typeahead: if minlength is 0 then display options on focus

### CODE REVIEW

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -295,6 +295,13 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
         }
       });
 
+      element.bind('focus', function (evt) {
+        if(minSearch == 0){
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Should be `===`

File: src/typeahead/typeahead.js [hunk 2]
```diff
@@ -295,6 +295,13 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
...
+          hasFocus = true;
+          getMatchesAsync(undefined);
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: Should execute `modelCtrl.$setViewValue(undefined)` beforehand.

Comment (wesleycho) [inline src/typeahead/typeahead.js]: Actually, change undefined to an empty string for this line and the suggested one to add beforehand.

**Comment by wesleycho** [comment]
This requires a unit test.

**Comment by rvanbaalen** [comment]
(By now) Travis is failing. Can you make it compatible with the latest UI Bootstrap?

**Comment by wesleycho** [comment]
Closing due to suffering from bitrot - if there is any interest in this feature, please open a new PR implementing the feature again with the appropriate unit tests.

**Pull request closed without merging**